app/services/transfer_destroy_service.rb

```diff
@@ -0,0 +1,29 @@
+class TransferDestroyService
+  def initialize(transfer_id:)
+    @transfer_id = transfer_id
+  end
+
+  def call
+    transfer.transaction do
+      revert_inventory_transfer!
+      transfer.destroy!
+    end
+
+    OpenStruct.new(success?: true)
+  rescue StandardError => e
+    OpenStruct.new(success?: false, error: e)
+  end
+
+  private
+
+  attr_reader :transfer_id
+
+  def transfer
+    @transfer ||= Transfer.find(transfer_id)
```

Comment:
Does this need to be `@transfer_id` or does the `attr_reader` call alias this to it?

Reply:
@mdworken the `attr_reader` aliases it. I generally try to avoid putting too many `@` variables as a preference and I find it cleaner :).